File: bench/Reference/binary_write_read.erl
#!/usr/bin/env escript
%% Read 5MB of binary, write it back, read again; print length.

main(_) ->
    {ok, F1} = file:open("bench/data/bench_binary.bin", [read, binary, raw]),
    {ok, Data} = file:read(F1, 5242880),
    ok = file:close(F1),
    ok = file:write_file("/tmp/erl_bench_binary_wr.bin", Data),
    {ok, Data2} = file:read_file("/tmp/erl_bench_binary_wr.bin"),
    io:format("~p~n", [byte_size(Data2)]).

File: bench/Reference/binary_write_read.py
#!/usr/bin/env python3
"""Python reference: binary file read + write + read-back"""

with open("bench/data/bench_binary.bin", "rb") as f:
    data = f.read()

with open("/tmp/py_bench_binary_wr.bin", "wb") as f:
    f.write(data)

with open("/tmp/py_bench_binary_wr.bin", "rb") as f:
    data2 = f.read()

print(len(data2))
